src/Repository/NphSampleRepository.php

```diff
@@ -47,4 +48,26 @@ public function findOneBySomeField($value): ?NphSample
         ;
     }
     */
+
+    public function findActiveSampleCodes(NphOrder $order, $site)
+    {
+        return $this->createQueryBuilder('s')
+            ->select('s.sampleCode')
+            ->join('s.nphOrder', 'o')
+            ->andWhere('o.participantId = :participantId')
+            ->andWhere('o.module = :module')
+            ->andWhere('o.timepoint = :timepoint')
+            ->andWhere('o.visitType = :visitType')
+            ->andWhere('o.site = :site')
+            ->andWhere('s.modifyType IN (:types) or s.modifyType is null')
+            ->setParameter('participantId', $order->getParticipantId())
```

Comment:
Instead of using setParameter for each param you could use setParameters and pass all these params as an assoc array.

`->setParameters(['participantId' => $order->getParticipantId(), 'module' => $order->getModule()]`